AwooChat.py

```diff
@@ -29,84 +29,76 @@ def titleWatchdog(msg):
-    p = prefs
 
-    now = int((datetime.now() - datetime(1970, 1, 1)).total_seconds() * 1000)
-    message = ""
+    now_time = int((datetime.now() - datetime(1970, 1, 1)).total_seconds() * 1000).time()
 
-    parsedPokemon = {}
+    pokemonBySource = {}
 
-    for chat_id in p.all():
-    
-        source = prefs.get(chat_id, 'pokemonDataSource', None)
-        if source is None:
-            continue
-            
-        source = source + "/raw_data"
-
-        s = json.load(urllib2.urlopen(source))
-        pokemonList = s["pokemons"]
-
-        for i in pokemonList:
-            id = i['pokemon_id']
-            lat = i['latitude']
-            lon = i['longitude']
-            enc_id = i['encounter_id']
-            despawn = datetime.fromtimestamp(i['disappear_time'] / 1000).strftime('%H:%M:%S')
-
-            if (enc_id not in pokemonWatchdogSeen):
-                parsedPokemon[id] = parsedPokemon.get(id, [])
-                parsedPokemon[id].append({'lat': lat, 'lon': lon, 'despawn': despawn})
-                pokemonWatchdogSeen.append(enc_id)
-                
-                
+    for chat_id in prefs.all():
         # Skip the program chats
         if chat_id == "__default" or chat_id == "__private" or chat_id == "__group":
             continue
-            
-        message = ""
-        alertList = p.get(chat_id, 'alertPokemonList')
 
-        if alertList is None:
+        source = prefs.get(chat_id, 'pokemonDataSource')
+        if not source:
+            continue
+
+        source += "/raw_data"
+
+        alertList = prefs.get(chat_id, 'alertPokemonList')
+        if not alertList:
             continue
 
-        for i in parsedPokemon.keys():
-            if i in alertList:
-                for j in parsedPokemon[i]:
-                    message += AwooUtils.getPokemonName(i) + " has been spotted! It despawns at " + str(j['despawn']) + \
-                            ". [ᕕ( ᐛ )ᕗ](http://maps.google.com?q=" + str(j['lat']) + "," + str(j['lon']) + ")\n"
-
-        if (chat_id < 0):
-            # Group chat, so we need to handle this
-            now_time = now.time()
-            if now_time >= time(06, 00) and now_time <= time(23, 59):
-                if len(message) != 0:
-                    if (len(message) > 4095):
-                        messages = message.split('\n')
+        if source not in pokemonBySource
+            pokemonBySource[source] = parsePokemon(source)
+        parsedPokemon = pokemonBySource[source]
+        message = ""
+
+        for i in set(alertList) & set(parsedPokemon.keys()):
+            for j in parsedPokemon[i]:
+                message += AwooUtils.getPokemonName(i) + \
+                        " has been spotted! It despawns at " + \
+                        str(j['despawn']) + \
+                        ". [ᕕ( ᐛ )ᕗ](http://maps.google.com?q=" + \
+                        str(j['lat']) + "," + str(j['lon']) + ")\n"
+
+        if len(message) != 0 and \
```

Comment:
There was a lot of duplicated logic and code here so I [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)'d it up to make it easier to read and maintain.